rust/connlib/tunnel/src/client.rs

```diff
@@ -191,7 +200,9 @@ where
         resource_id: ResourceId,
         gateway_id: GatewayId,
         relays: Vec<Relay>,
+        site_id: SiteId,
     ) -> connlib_shared::Result<Request> {
+        self.role_state.gateways_site.insert(gateway_id, site_id);
```

Comment:
Please no direct field access here. The functions on `ClientState` and `GatewayState` should be the real API. Everything in `Tunnel` is just glue-code and should not contain any functionality other than delegating.